hr_holidays_public/models/resource_calendar.py

```diff
@@ -15,20 +15,42 @@ class ResourceCalendar(models.Model):
     def _attendance_intervals_batch_exclude_public_holidays(
         self, start_dt, end_dt, intervals, resources, tz
     ):
-        list_by_dates = (
-            self.env["hr.holidays.public"]
-            .get_holidays_list(
-                start_dt=start_dt.date(),
-                end_dt=end_dt.date(),
-                employee_id=self.env.context.get("employee_id", False),
+        holidays_by_country = {
+            False: set(
+                self.env["hr.holidays.public"]
+                .get_holidays_list(
+                    start_dt=start_dt.date(),
+                    end_dt=end_dt.date(),
+                    employee_id=self.env.context.get("employee_id", False),
+                )
+                .mapped("date")
             )
-            .mapped("date")
-        )
+        }
+        if resources:
+            employees = self.env["hr.employee.public"].search(
```

Comment:
hello @gurneyalex 
on one of projects we encounter empty recordset for `employees`, which can cause further mapping issues (KeyError) looking for the country
My suggestion is to improve little bit `resource in resources` loop:
```
for resource in resources:
    interval_resource = intervals.get(resource.id)
    attendances = []
    country = resource_country.get(resource.id, self.env["res.country"])
    holidays = holidays_by_country.get(country)
    for attendance in interval_resource._items:
        if holidays and attendance[0].date() not in holidays:
            attendances.append(attendance)
    intervals[resource.id] = Intervals(attendances)
```
This is checked on actual database